### PR TITLE
Added migrator for aarch64 and ppc64le

### PR DESCRIPTION
**Pull request by mariusvniekerk**
This is a basic migrator for doing the Arch rebuild.

* We are purposefully ignoring noarch here and initially operating within a bounded set of packages.
* In order to determine if a package is migrated we take a look inside the conda-forge-config.yml and check for certain values.  If those are present a package is skipped.  
* The current set of constraints yields about 120 packages to run this migrator on.

Additionally, this also adds a refactor to allow using `distributed` instead of `multiprocessing` for performing the initial graph building.

### CODE REVIEW

File: conda_forge_tick/auto_tick.xsh [hunk 1]
```diff
@@ -206,6 +207,39 @@ def add_rebuild(migrators, gx):
...
+    total_graph.remove_edges_from(total_graph.selfloop_edges())
+
+    top_level = {node for node in total_graph if not set(total_graph.predecessors(node))}
+    cycles = list(nx.simple_cycles(total_graph))
+    print('cycles are here:', cycles)
+
+    migrators.append(
+        ArchRebuild(graph=total_graph,
+                pr_limit=5,
+                name='aarch64 and ppc64le addition',
```
[the remark this answers is not in the document]
Comment (isuruf) [reply conda_forge_tick/auto_tick.xsh]: Appveyor does. Travis-CI can be implemented using conditions. For circle, we can modify fast-finish script

File: conda_forge_tick/auto_tick.xsh [hunk 2]
```diff
@@ -26,6 +26,7 @@ $MIGRATORS = [
    # Noarch(pr_limit=10),
    # Pinning(pr_limit=1, removals={'perl'}),
    # Compiler(pr_limit=7),
+   ArchRebuild(pr_limit=7)
```
Comment (CJ-Wright) [inline conda_forge_tick/auto_tick.xsh]: I don't think you need this here, since you are adding it below.

File: conda_forge_tick/migrators.xsh [hunk 3]
```diff
@@ -711,6 +697,83 @@ class Rebuild(Migrator):
...
+    def commit_message(self):
+        return "add noarch r"
+
+    def pr_title(self):
+        if self.name:
+            return 'Noarch R for ' + self.name
+        else:
+            return 'Bump build number'
+
+    def pr_head(self):
```
Comment (CJ-Wright) [inline conda_forge_tick/migrators.xsh]: I think you might get this from the superclass

**Comment by scopatz** [comment]
Seems to be failing CI

**Comment by mariusvniekerk** [comment]
ah its failing because of the moved out r stuff

File: conda_forge_tick/auto_tick.xsh (hunk 1 above)
[the remark this answers is not in the document]
Comment (isuruf) [reply conda_forge_tick/auto_tick.xsh]: Can we at least do `[skip appveyor]` before we do this?

[the remark this answers is not in the document]
Comment (CJ-Wright) [reply conda_forge_tick/auto_tick.xsh]: Would you mind putting in a PR?

[the remark this answers is not in the document]
Comment (isuruf) [reply conda_forge_tick/auto_tick.xsh]: I opened a issue at https://github.com/regro/cf-scripts/issues/452

**Pull request merged**